File: sorted.js
const C_TIMES = 5;
const C_MILLIONS = [1, 1.5, 3, 6];

/** really inneficient because destructuring and spread operations...
const sorted = ([pivot, ...others]) => pivot === void 0 ? [] : [
  ...sorted(others.filter(x => x < pivot)),
  pivot,
  ...sorted(others.filter(x => x >= pivot))
];
*/

const sorted = (xs)=> xs.length===0 ? [] : [].concat(
  sorted( xs.slice(1).filter(x=> x<xs[0]) ),
  xs[0],
  sorted( xs.slice(1).filter(x=> x>=xs[0]) )
);


C_MILLIONS.forEach(millions =>
  test(C_TIMES, millions * 1000000)
);

function test(times, size) {
  let total = 0;
  for (let n = 0; n < times; n++) {
    const numbers = [...Array(size).keys()].map(i => Math.random());
    total += measure(() => sorted(numbers));
  }
  console.log(`${size},${times > 0 ? Math.round(total / times) : 0}`);
}

function measure(f) {
  const t_ini = Date.now();
  f();
  return Date.now() - t_ini;
}

File: sorted.jl
using Printf

C_TIMES = 5
C_MILLIONS = [1, 1.5, 3, 6]

function sorted(xs::Array{Float64, 1})::Array{Float64, 1}
  return length(xs) == 0 ? Array{Float64, 1}() : vcat(
    sorted([ x for x in xs[2:end] if x < xs[1] ]),
    xs[1],
    sorted([ x for x in xs[2:end] if x >= xs[1] ])
  )
end

function test(times, size)
  total = 0;
  for n = 1:times
    numbers::Array{Float64, 1} = [ rand() for a = 1:size ]
    r = @timed begin
      sorted(numbers)
    end
    total = total + r[2] * 1000
  end
  @printf "%d,%d\n" size (times>0 ? total/times : 0)
end

for millions in C_MILLIONS
  test(C_TIMES, millions * 1000000)
end